aptos-move/parallel-executor/src/errors.rs

```diff
@@ -1,16 +1,16 @@
 // Copyright (c) Aptos
 // SPDX-License-Identifier: Apache-2.0
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum Error<E> {
     /// Invariant violation that happens internally inside of scheduler, usually an indication of
     /// implementation error.
     InvariantViolation,
-    /// The inference can't get the read/write set of a transaction, abort the entire execution pipeline.
-    InferencerError,
-    /// A transaction write to a key that wasn't estimated by the inferencer, abort the execution
-    /// because we don't have a good way of handling read-after-write dependency. Will relax this limitation later.
-    UnestimatedWrite,
+    /// The same module access path for module was both read & written during speculative executions,
```

Comment:
I don't see elsewhere mentioned, but the fact the read/write sets are transitive over dependencies is important for the correctness and should likely be mentioned somewhere.

I also see that you do not use the new `get_and_clear_module_cache_hits`. I know we discussed you don't need it, but it might be good to document somewhere in the code the entire reasoning. Given the VMs cache behavior, I think for any set of transactions {T_1, T_2, ...} executed on the same cache, if T_i has a cache hit so you wont see any module read, there must be T_j (i != j) which exposed the module read.

Well at least after writing this down I feel a bit surer again it is correct ;-)

Reply:
good point, thanks! clarified a bit with the error and added a comment to the record function that detects the intersection.